src/backend/users/management/commands/migrate_housing_seeker_data.py

```diff
@@ -0,0 +1,488 @@
+                    has_voucher=has_voucher,
+                    voucher_number=voucher_number,
+                    voucher_bedrooms=voucher_bedrooms,
+                    travel_mode=travel_mode,
+                    commute_priority=commute_priority,
+                    public_safety_priority=public_safety_priority,
+                    school_quality_priority=school_quality_priority,
+                )
+                user_profile.save()
+                self.added_profiles += 1
+        else:
+            self.skipped_user_profile_key_s3.append({"key": key, "reason": reason})
+            self.stdout.write(f"{reason}. Skipping user profile with key: {key}.\n")
+
+        return user_profile
+
+    def has_location_fields(self, destination):
+        reason = ""
+        if DestinationKeys.LOC not in destination:
+            reason += "NO LOCATION; "
+            return False, reason
+        loc = destination[DestinationKeys.LOC]
+        if LocationKeys.POSITION not in loc:
+            reason += "NO POSITION; "
+            return False, reason
+        pos = loc[LocationKeys.POSITION]
+        if PositionKeys.LAT not in pos or PositionKeys.LON not in pos:
+            reason += "NO FULL COORDINATE PAIR; "
+            return False, reason
+
+        return True, reason
+
+    def decode_purpose(self, purpose):
+        if purpose.lower() == "friends":
+            return Destination.TripPurpose.FRIENDS
+        if purpose.lower() == "work":
+            return Destination.TripPurpose.WORK
+        if purpose.lower() == "doctor":
+            return Destination.TripPurpose.DOCTOR
+        if purpose.lower() == "day care":
+            return Destination.TripPurpose.DAYCARE
+        if purpose.lower() == "family":
+            return Destination.TripPurpose.FAMILY
+        if purpose.lower() == "worship":
+            return Destination.TripPurpose.WORSHIP
+
+        return Destination.TripPurpose.OTHER
+
+    def save_destination(self, user_profile, des):
+        destination = None
+        reason = ""
+
+        has_location, has_loc_reason = self.has_location_fields(des)
+        has_purpose = self.is_key_val_nonempty(des, DestinationKeys.PURPOSE)
+        has_primary = self.is_key_val_nonempty(des, DestinationKeys.PRIMARY)
+
+        if not has_location:
+            reason += has_loc_reason
+        if not has_purpose:
+            reason += "NO PURPOSE; "
+        if not has_primary:
+            reason += "NO PRIMARY; "
+
+        # when any of the required field is empty
+        # don't attach the corresponding destination to user profile
+        if has_location and has_purpose and has_primary:
+            lat = des[DestinationKeys.LOC][LocationKeys.POSITION][PositionKeys.LAT]
+            lon = des[DestinationKeys.LOC][LocationKeys.POSITION][PositionKeys.LON]
+            purpose = self.decode_purpose(des[DestinationKeys.PURPOSE])
+
+            destination = Destination(
```

Comment:
If you want to create and save in one step you can use `Model.objects.create()` instead of `obj = Model(); obj.save()`

Reply:
Oh, good point! This is actually very good to know and less verbose. I think the reason why I passed the result to `destination` is because I'd like to return it in the end. If the return is not `None`, then it won't be reported as a part of the summary in the end. What do you think?

Reply:
No worries! Addressed here: https://github.com/azavea/echo-locator/pull/478/commits/a36a8f148ac0ae9312f701d016466514c1a7c864